app/models/user.rb

```diff
@@ -0,0 +1,30 @@
+class User < ActiveRecord::Base
+  has_secure_password
+  has_many :questions
+  has_many :answers
+  has_many :comments
+  has_many :votes
+
+  def reputation
+    question_rep = self.questions.map { |q| q.score }.inject(:+)
+    answer_rep = self.answers.map { |a| a.score }.inject(:+)
+    comment_rep = self.comments.map {|c| c.score }.inject(:+)
+    question_rep + answer_rep + comment_rep
+  end
+
+  def owns?(submission_type, submission_id)
+    case submission_type
```

Comment:
You could pass submission as a single param and read
submission.class.name and submission.id inside the method.